lib/index.js

```diff
@@ -0,0 +1,55 @@
+import createLogger from './logs';
+import createKey from './key';
+import pandaValidate from 'pan-domain-validate-user';
+
+const AUTH_FAILED_ERROR_MESSAGE = 'panda-auth-failed';
+const COOKIE_NAME = 'gutoolsAuth-assym';
+
+export default function(config, overrideHttps, validate = pandaValidate) {
+    // For backward compatibility, config can be the domain string or an object
+    if (typeof config === 'string') {
+        config = { domain: config };
+    }
+    const {
+        domain,
+        refresh = 10 * 60 * 1000, // 10 minutes
```

Comment:
I've changed it to 2 minutes.

On every request, the code checks the cached key, if it's older than 2 minutes it downloads it again. It's not a background polling.